storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java

```diff
@@ -361,7 +364,11 @@ public void doHeartBeat() throws IOException {
         state.setWorkerHeartBeat(lsWorkerHeartbeat);
         state.cleanup(60); // this is just in case supervisor is down so that disk doesn't fill up.
         // it shouldn't take supervisor 120 seconds between listing dir and reading it
-        heartbeatToMasterIfLocalbeatFail(lsWorkerHeartbeat);
+
+        if (!workerState.stormClusterState.isPacemakerStateStore()) {
+            LOG.debug("If pacemaker is not used, send supervisor");
```

Comment:
We should delete `If`. Since if this message is logged, it means pacemaker is not being used.

Reply:
Addressed